### PR TITLE
Modify check_b0_threshold and typo fix

### PR DESCRIPTION
**Pull request by boiteaclou**
Add overwrite and assert on outputs to extract_b0 and fix typo in scil_merge_sh

### CODE REVIEW

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by boiteaclou** [comment]
Check b0 threshold has been updated, it is less restrictive than the previous implementation and returns the valid threshold in regards of the parameters.

- Before, if a bval_min was over the DEFAULT_B0_THRESHOLD, the execution would end in error. Now, it is still the case if no custom threshold is passed in the function arguments, of if the bval_min is over the custom threshold
- Before, if a bval_min was negative, it would end in error. Now it's only a warning (not 100% sure about that behavior, could add error if no force_b0_threshold)
- If the custom threshold passed is over DEFAULT_B0_THRESHOLD, a warning message is printed, but execution continues

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Review by arnaudbore — APPROVED** [review]
Typo otherwise LGTM

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Review by arnaudbore — APPROVED** [review]

File: scilpy/utils/bvec_bval_tools.py [hunk 1]
```diff
@@ -65,44 +65,60 @@ def normalize_bvecs(bvecs, filename=None):
...
+            logging.warning(
+                'Warning: Your minimal bvalue is {}, but the threshold '
+                'is set to {}. Since --force_b0_threshold was specified, '
+                'the script will proceed with a threshold of {}'
+                '.'.format(bvals_min, b0_thr, bvals_min))
+            return bvals_min
         else:
-            logging.warning('Warning: No b=0 image. Setting b0_threshold to '
-                            'the minimum bval: {}'.format(bvals_min))
+            raise ValueError('The minimal bvalue is greater than the '
```
Comment (arnaudbore) [inline scilpy/utils/bvec_bval_tools.py:112]:
```suggestion
            raise ValueError('The minimal bvalue ({}) is greater than the '
```

File: scilpy/utils/bvec_bval_tools.py [hunk 2]
```diff
@@ -65,44 +65,60 @@ def normalize_bvecs(bvecs, filename=None):
...
+                             'threshold ({}). No b0 volumes can be found.\n'
+                             'Please check your data to ensure everything '
+                             'is correct.\n'
+                             'Value found: {}\n'
```
Comment (arnaudbore) [inline scilpy/utils/bvec_bval_tools.py:116]:
```suggestion

```

**Review by arnaudbore — APPROVED** [review]

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**